core/task.go

```diff
@@ -91,6 +91,15 @@ type Task interface {
 
 type TaskOption func(Task) TaskOption
 
+type A int
+
+type B struct {
+	foo int
+	bar int
+}
+
+type x func(int) x
+
```

Comment:
It is not! You'll find that the later iterations of this pattern don't use
it, e.g. options on plugin meta.
On Wed, Mar 9, 2016 at 13:47 Pawel Palucki <notifications@github.com> wrote:

> In core/task.go
> <https://github.com/intelsdi-x/snap/pull/757#discussion_r55595617>:
>
> > @@ -91,6 +91,15 @@ type Task interface {
> >
> >  type TaskOption func(Task) TaskOption
> >
> > +type A int
> > +
> > +type B struct {
> > +	foo int
> > +	bar int
> > +}
> > +
> > +type x func(int) x
> > +
>
> argh, I was trying to explain my coworker, how the TaskOption (undoing)
> works - the recursive func type - and forget to remove example - I'll
> delete it
> ps. is "undoing" feature used somewhere ? I couldn't find any usage -
> return value from calling is always ignored - or I'm missing something
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/intelsdi-x/snap/pull/757/files#r55595617>.
>